src/main/java/Duke.java

```diff
@@ -1,10 +1,275 @@
+        } else {
+            addTask(input);
+        }
+    }
+
+    private static void markItem(String[] words) {
+        int indexToMark = getIndex(words);
+        if (isIndexValid(indexToMark)) {
+            list[indexToMark].markAsDone();
+            printMarkIsCompleted(list[indexToMark]);
+        } else {
+            handleInvalidIndex();
+        }
+    }
+
+    private static void unmarkItem(String[] words) {
+        int indexToUnmark = getIndex(words);
+        if (isIndexValid(indexToUnmark)) {
+            list[indexToUnmark].markAsUndone();
+            printUnmarkIsCompleted(list[indexToUnmark]);
+        } else {
+            handleInvalidIndex();
+        }
+    }
+
+    private static void handleInvalidIndex() {
+        System.out.println(INVALID_INDEX);
+        System.out.println(LINE);
+    }
+
+    private static boolean isIndexValid(int indexToUnmark) {
+        boolean isAboveZero = indexToUnmark >= MINIMUM_INDEX;
+        boolean isBelowMaximum = indexToUnmark < index;
+        return isAboveZero && isBelowMaximum;
+    }
+
+    private static int getIndex(String[] words) {
+        return Integer.parseInt(words[1]) - 1;
+    }
+
+    private static void addTask(String input) {
+        String typeOfTask = getTypeOfTask(input);
+        if (isToDo(typeOfTask)) {
+            addToDo(input);
+        } else if (isDeadline(typeOfTask)) {
+            addDeadline(input);
+        } else if (isEvent(typeOfTask)) {
+            addEvent(input);
+        } else {
+            addNormalTask(input);
+        }
+        printConfirmation();
+        incrementIndex();
+    }
+
+    private static void incrementIndex() {
+        index++;
+    }
+
+    private static void printConfirmation() {
+        int numberOfTasks = index + 1;
+        System.out.println(ADDED);
+        System.out.println(list[index]);
+        System.out.println(NUMBER_OF_TASKS_FIRST_HALF + numberOfTasks + NUMBER_OF_TASKS_SECOND_HALF);
+        System.out.println(LINE);
+    }
+
+    private static void addNormalTask(String input) {
+        list[index] = new Task(input);
+    }
+
+    private static String getTypeOfTask(String input) {
+        String[] words = input.split(SPACEBAR);
+        return words[0];
+    }
+
+    private static void addEvent(String input) {
+        int indexOfAt = getIndexOfAt(input);
+        String description = getDescriptionOfEvent(input, indexOfAt);
+        String at = getTimingOfEvent(input, indexOfAt);
+        list[index] = new Event(description, at);
+    }
+
+    private static String getTimingOfEvent(String input, int indexOfAt) {
+        int length = input.length();
+        String at = input.substring(indexOfAt + START_INDEX_OF_TIMING, length);
+        return at;
+    }
+
+    private static String getDescriptionOfEvent(String input, int indexOfAt) {
+        String description = input.substring(START_INDEX_OF_EVENT, indexOfAt - LENGTH_OF_SPACE);
+        return description;
+    }
```

Comment:
similar methods as addDeadline getTimingOfDeadline, and getDescriptionOfDeadline, maybe you can abstract it out in the TaskManager class as well.